src/rules-schema.json

```diff
@@ -0,0 +1,59 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"type": "object",
+	"properties": {
+		"rules": {
+			"type": "array",
+			"items": {
+				"type": "object",
+				"properties": {
+					"name": {
+						"type": "string"
+					},
+					"conditions": {
+						"type": "array",
+						"items": {
+							"type": "object",
+							"properties": {
+								"type": {
+									"type": "string",
+									"enum": ["inCategory"]
+								},
+								"categories": {
+									"type": "array",
+									"items": {
+										"type": "string"
+									},
+									"minItems": 1
+								}
+							},
+							"required": ["type", "categories"],
+							"additionalProperties": false
+						}
+					},
+					"actions": {
+						"type": "array",
+						"items": {
+							"type": "object",
+							"properties": {
+								"type": {
+									"type": "string",
+									"enum": ["addCategory"]
+								},
+								"category": {
+									"type": "string"
+								}
+							},
```

Comment:
In the future if we have different action types defined in a registry, then we're either going to end up with a single big schema with all the possible rules, or else the schema is going to be very basic and the type-specific validation will likely need to be done directly in PHP.